### ISSUE TITLE
Unable to launch rmf office demo properly

### ISSUE THREAD

**Issue by JGUO006** [open-rmf/rmf#92]
Hi,

I just followed instructions to download and compile the rmf_ws source code today.

The downloading and compiling were smooth and successful.

However, when I ran the first rmf_demo on office map, there were no robots spawn on Rviz and Gazebo was also not launched.

In RMF panel, there was no robot's information. Tasks could be submitted but had no action from the system.

Some screen captures when I launch the demo are as follow,

![Terminal_1](https://user-images.githubusercontent.com/78726874/131812947-f048c145-d667-4d4b-809c-e3e54f04fc97.png)

![Terminal_2](https://user-images.githubusercontent.com/78726874/131812970-a315a7b3-aef9-46db-977b-65b4e2547524.png)

![Terminal_3](https://user-images.githubusercontent.com/78726874/131812997-0ed62f94-2362-43b1-9f61-0ee0a4cf0e65.png)

![Rviz](https://user-images.githubusercontent.com/78726874/131810859-757dffaa-0f33-4ac7-9f3e-5dad689c350c.png)

![RMF_Panel](https://user-images.githubusercontent.com/78726874/131810891-29bdd3f2-5348-4e86-93e1-2e9e3d290f65.png)

I am using Ubuntu 20.04 with both ROS noetic and ROS 2 foxy installed.

Do you have any idea what issue is causing that?

**Comment by marcoag** [comment]
I see the error "Address already in use" for `localhost:8080`, can you check you don't have anything running on `localhost` at that port? it seems it might be conflicting...

**Comment by aaronchongth** [comment]
It would be helpful if you could copy out the log into a github gist too, thanks!

**Comment by Yadunund** [comment]
@JGUO006 as of PR https://github.com/open-rmf/rmf_demos/pull/80, the correct way to launch the demo is ```ros2 launch rmf_demos_gz office.launch.xml```. But in your first screenshot, you're using the old command ```ros2 launch rmf_demos office.launch.xml```
This was done to separate `gazebo` and `ignition` dependencies in launching the demos.

**Comment by Yadunund** [comment]
Sorry, I accidentally closed this ticket. My suggestion above is relevant to not starting gazebo without which, there are no robots that make their presence known to the fleet adapters. As a result, the adapters will not respond to the bid for a new task. But as @marcoag and @aaronchongth have pointed out, there is another issue with the conflicting addresses on your system.

**Comment by JGUO006** [comment]
Hi,

Thanks all for your prompt reply! I have checked that one of my docker containers installed recently is using port 8080. It is causing the port conflicting (already in use) issue.

After I stopped docker containers and checked that the port 8080 is not being used any more, I ran the old command ros2 launch rmf_demos office.launch.xml first. There was no error message but the result was the same as mentioned above. No robots spawn and Gazebo not pop up.

Then I ran the recommended command ros2 launch rmf_demos_gz office.launch.xml, everything was just back to normal.

Here are some screen captures,

![Terminal_4](https://user-images.githubusercontent.com/78726874/131829834-02c0e01c-059a-44d4-a28b-f9da232fce71.png)

![RMF_Panel_2](https://user-images.githubusercontent.com/78726874/131829893-fe0cbb50-164e-4399-8989-d02ac931fb61.png)

![Rviz_2](https://user-images.githubusercontent.com/78726874/131829915-8c42d6c6-a10c-4913-ab9a-474b95024279.png)

![Gazebo_2](https://user-images.githubusercontent.com/78726874/131829933-f2aba555-c4c2-4482-b446-4997015c88dd.png)

That solved this issue here. Thank you!
